kernel/arch/riscv64/Hat.cpp

```diff
@@ -0,0 +1,250 @@
+#include <arch/Hat.h>
+#include <memory/Pmm.h>
+#include <debug/Log.h>
+#include <Memory.h>
+#include <Maths.h>
+
+namespace Npk
+{
+    enum PageSizes
+    {
+        _4K = 1,
+        _2M = 2,
+        _1G = 3,
+        _512G = 4,
+    };
```

Comment:
just a question, not a review comment: why rewriting the same enum two times? can't it be shared by different builds?

Reply:
Fair question: you're right, it could. There's no technical reason not to.
My thinking was for isolation, these definitions are specific to the mmu of x86/riscv and it just so happens that their paging mechanisms are quite similar and I ended up with 2 similar enums. The riscv enum could be slightly different as 5 level paging allows for 256TiB pages as well, but I dont intend to support those. My plan was for anything shared across architectures to be in the header file (Hat.h - which is used by the VMM) while platform-specific details like this are kept away from the rest of the kernel, so I dont accidentally rely on platform-specific features if I decide to add another architecture in the future.
I could probably move this enum and some other things into a header that is included by both platforms when I think about it.